src/geometry/collider.rs

```diff
@@ -49,6 +49,8 @@ impl ColliderFlags {
 pub struct Collider {
     shape: SharedShape,
     density: Real,
+    /// If None, use [`Self::density`] and [`SharedShape::mass_properties`].
+    mass_properties: Option<Box<MassProperties>>,
```

Comment:
Since both `density` and `mass_properties` are exclusive, let's make this explicit by using [either](https://crates.io/crates/either): `density_or_mass_properties: Either<Real, Box<MassProperties>>`

That way we are sure that we won't forget to use the correct one, and we don't have to do all the `mass(&mp) / volume` computation added by this PR.

This will require one breaking change: `Collider::density()` will have to return an `Option<Real>`, indicating that its `MassProperties` have been set explicitly.